src/block/block_tiered.c

```diff
@@ -0,0 +1,94 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_block_tiered_flush --
+ *     Flush this file, start another file.
+ */
+int
+__wt_block_tiered_flush(
+  WT_SESSION_IMPL *session, WT_BLOCK *block, uint8_t **flush_cookie, size_t *cookie_size)
+{
+    /* TODO: tiered: fill in the cookie. */
+    (void)flush_cookie;
+    (void)cookie_size;
+
+    return (__wt_block_tiered_newfile(session, block));
+}
+
+/*
+ * __wt_block_tiered_load --
+ *     Set up log-structured processing when loading a new root page.
+ */
+int
+__wt_block_tiered_load(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_BLOCK_CKPT *ci)
+{
+    /*
+     * TODO: tiered: this call currently advances the object id, that's probably not appropriate for
+     * readonly opens. Perhaps it's also not appropriate for opening at an older checkpoint?
+     */
+    if (block->log_structured) {
+        block->logid = ci->root_logid;
+
+        /* Advance to the next file for future changes. */
+        WT_RET(__wt_block_tiered_newfile(session, block));
+    }
+    return (0);
+}
+
+/*
+ * __wt_block_tiered_newfile --
+ *     Switch a log-structured block object to a new file.
+ */
+int
+__wt_block_tiered_newfile(WT_SESSION_IMPL *session, WT_BLOCK *block)
+{
+    WT_DECL_ITEM(tmp);
+    WT_DECL_RET;
+    WT_STORAGE_SOURCE *storage_source;
+    const char *filename;
+
+    /* Get the old file name again. */
+    WT_ERR(__wt_scr_alloc(session, 0, &tmp));
+    WT_ERR(__wt_buf_fmt(session, tmp, "%s.%08" PRIu32, block->name, block->logid));
```

Comment:
I think this whole thing needs a big TODO correct? `tiered_switch` is generating the `object:example.00000000#.wtobj` name and I think the name in the storage should be whatever it is given `example.000000000#.wtobj` right?

Reply:
I think `block->logid` is not needed and should go away, right? The "source of truth" for the next object number is `tiered->next_id` and `tiered->current_id`. Should this use that now?

Reply:
I added TODOs for this.  This change is getting large and merges are getting hard, so I'd rather not do it now.